### ISSUE TITLE
Incorrect line numbers output for Unused Variable errors

### ISSUE THREAD

**Issue by paulschreiber** [Automattic/VIP-Coding-Standards#446]
## Bug Description

Incorrect line numbers are output for Unused Variable errors.

## Minimal Code Snippet

```php
<?php
function foo( $html ) {
	list( $red, $green, $blue ) = something( $html );
	count( $red );
}
```

## Error Code

```
$ vendor/bin/phpcs --severity=1  ~/test.php 
W 1 / 1 (100%)



FILE: Users/schreipa/test.php
---------------------------------------------------------------------------------------------------------
FOUND 0 ERRORS AND 2 WARNINGS AFFECTING 1 LINE
---------------------------------------------------------------------------------------------------------
 1 | WARNING | Unused variable `$green`.
   |         | (WordPressVIPMinimum.Variables.VariableAnalysis.UnusedVariable)
 1 | WARNING | Unused variable `$blue`.
   |         | (WordPressVIPMinimum.Variables.VariableAnalysis.UnusedVariable)
---------------------------------------------------------------------------------------------------------

Time: 295ms; Memory: 10MB
```

## Environment

Use `php -v` and `composer show` to get versions.

| Question                | Answer
| ------------------------| -------
| PHP version             | 7.3.3
| PHP_CodeSniffer version | 3.4.2
| VIPCS version           | 2.0.0

## Additional Context (optional)
Enable these rulesets:
```
	<rule ref="WordPress-Extra" />
	<rule ref="WordPress-VIP-Go" />
```

**Comment by GaryJones** [comment]
Good catch, and thanks for reporting.

The version of VariableAnalysis we have is somewhat out of date - https://github.com/Automattic/VIP-Coding-Standards/issues/236 is the ticket to resolve this, and I'd like to find time with @sirbrillig on the best way to get VIPCS using the up to date version.

I'll leave this ticket open for now, as a reminder to check that this particular edge case is addressed once we have the new version of VA, but it's honestly unlikely to get addressed directly in the meantime.

**Comment by sirbrillig** [comment]
👋 @GaryJones I'm happy to go over this together some time! Maybe ping me on Slack and we can chat about it in more detail?

**Comment by paulschreiber** [comment]
Ran in to this again. Any luck with your investigation?

**Comment by paulschreiber** [comment]
Hit this again today with liveblog.

**Comment by sirbrillig** [comment]
Looks like the discussion on this has moved to https://github.com/Automattic/VIP-Coding-Standards/issues/449 

(Please let me know if I can help, VIP friends!)

**Comment by GaryJones** [comment]
```
FILE: /Users/gary/code/phpcs-test.php
-------------------------------------------------------------------------------------------------------
FOUND 0 ERRORS AND 2 WARNINGS AFFECTING 1 LINE
-------------------------------------------------------------------------------------------------------
 3 | WARNING | Unused variable $green.
   |         | (VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable)
 3 | WARNING | Unused variable $blue.
   |         | (VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable)
-------------------------------------------------------------------------------------------------------

Time: 112ms; Memory: 6MB
```

Now that #449 is completed to use VariableAnalysis proper, I've confirmed that the provided example above doesn't give an error with the `develop` version of `WordPress-VIP-Go`, so this is now fixed for VIPCS 2.2.0.

Also, note #498 where we may silence the violations for unused variables completely.
